tensorflow/core/kernels/non_max_suppression_op.cc

```diff
@@ -143,13 +146,13 @@ void DoNonMaxSuppressionOp(
   std::priority_queue<Candidate, std::deque<Candidate>, decltype(cmp)>
       candidate_priority_queue(cmp);
   for (int i = 0; i < scores_data.size(); ++i) {
-    if (scores_data[i] > score_threshold) {
+    if (scores_data[i] > static_cast<T>(score_threshold)) {
```

Comment:
would it be better to upcast scores_data[i] to float.

Reply:
Thanks @azaks2. The PR has been updated to use upcast instead.